decentraland/unity-renderer
# Added profanity explicit words

## What does this PR change?

Changes the profanity chat filtering regex to add two types of offensive words:
- explicit: the entire word must be written exactly as it is, example "ass" so you can write "assembly" without being filtered
- non-explicit: the word must be contained in the phrase somehow: "shit123hehe" will be filtered as "****123hehe"

Also the offensive word list has been refined.
The list of words file is at `Assets/Scripts/MainScripts/DCL/Controllers/HUD/ChatWidgetHUD/Resources/Profanity/badwords.json`. You can check which words are explicit and which are not.

## How to test the changes?

1. Go to: https://play.decentraland.zone/?renderer-branch=feat/explicit-profanity-words
2. Go to settings menu, at general tab, you should have "Profanity Filter" toggle enabled.
3. Write any badword like "shithole im filtered" in the general chat. You should see the message in the chat like this: "****hole im filtered"
4. Write any private message to a friend that contains a badword like "im thinking with my ass". You should see the message unfiltered (private chat window and/or general chat)
5. Go to settings menu, at general tab, disable "Profanity Filter" toggle.
6. Repeat step 3 and 4. None of those messages should be filtered.

## Our Code Review Standards

https://github.com/decentraland/unity-renderer/blob/master/docs/code-review-standards.md
